### PR TITLE
add test for unicode paths

### PR DESCRIPTION
**Pull request by ObiWahn**
http://jenkins01.arangodb.biz:8080/view/Windows/job/arangodb-ANY-windows-matrix/74/

### CODE REVIEW

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/4332/

File: tests/js/server/paths/filesystem.js [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+    fs.write(tfn, "hello");
+    assertTrue(fs.isFile(tfn));
+    fs.remove(tfn);
+  }
+
+  function tryMakeAbsoluteAllowed(fn) {
+    fs.makeAbsolute(fn);
+  }
+  function tryCopyFileAllowed(sn, tn) {
+    try {
```
Comment (jsteemann) [inline tests/js/server/paths/filesystem.js]: this try...catch here is just unnecessary, and will obfuscate the stack trace by moving the error location from the `fs.copyFile()` to `assertTrue(false)`.

File: tests/js/server/paths/filesystem.js [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+      fs.copyFile(sn, tn);
+    } catch (err) {
+      assertTrue(false, "failed to copy " + sn + " to " + tn + " - " + err);
+    }
+    tryExistsAllowed(sn, true);
+    tryExistsAllowed(tn, true);
+  }
+  function tryCopyRecursiveFileAllowed(sn, tn) {
+    try {
```
Comment (jsteemann) [inline tests/js/server/paths/filesystem.js]: same (also true for the next few occurrences)

File: tests/js/server/paths/filesystem.js [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+////////////////////////////////////////////////////////////////////////////////
+
+const fs = require('fs');
+const internal = require('internal');
+
+let rootDir = fs.join(fs.getTempPath(), '..');
+let testFilesDir = rootDir;
+
+
+const topLevelAllowed = fs.join(testFilesDir, 'allowed');
```
Comment (jsteemann) [inline tests/js/server/paths/filesystem.js]: I would have expected the explicit usage of Unicode paths _here_.

Comment (ObiWahn) [reply tests/js/server/paths/filesystem.js]: Whe whole suite works in a Unicode path, so everything containing the root directory as prefix is tested with unicode strings. Do we gain anything by making it explicit?

File: tests/js/server/paths/filesystem.js (hunk 2 above)
Comment (dothebart) [reply tests/js/server/paths/filesystem.js]: the reason I did this originally is to easily have the combination of source file, destination file and error message in one string. Else you may need to search and (errnously) calculate the filename you believe that was copied.

NB: this function isn't used below.

File: tests/js/server/paths/filesystem.js [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+      });
+      fs.zipFile(zip, sourceDir, files);
+    } catch (err) {
+      assertTrue(false, "failed to Zip into " + zip + " these files: " + sourceDir + "[ " + files + " ] - " + err);
+    }
+    tryExistsAllowed(zip, true);
+  }
+
+  function tryUnZipFileAllowed(zip, sn) {
+    try {
```
Comment (jsteemann) [inline tests/js/server/paths/filesystem.js]: These try...catch wrappers just distract from the source of error.
Whenever one of the file operations fails, it can clearly be figured out from the stack trace which file/test caused the error.
Adding an extra framework around the assertions of the test framework is just clutter IMHO.

File: lib/V8/v8-shell.cpp [hunk 5]
```diff
@@ -266,9 +267,15 @@ static void JS_ProcessJsonFile(v8::FunctionCallbackInfo<v8::Value> const& args)
   // extract the callback
   v8::Handle<v8::Function> cb = v8::Handle<v8::Function>::Cast(args[1]);
 
-  // read and convert
-  std::string line;
+// read and convert
```
Comment (jsteemann) [inline lib/V8/v8-shell.cpp]: wrong indentation

File: lib/V8/v8-shell.cpp [hunk 6]
```diff
@@ -266,9 +267,15 @@ static void JS_ProcessJsonFile(v8::FunctionCallbackInfo<v8::Value> const& args)
...
+std::string line;
+#ifndef _MSC_VER
   std::ifstream file(*filename);
+#else
+  std::ifstream file;
```
Comment (dothebart) [inline lib/V8/v8-shell.cpp]: imho we should use something from basics to do this. We have a porting layer, we should not distribute #if wintendo all over the place.

Comment (ObiWahn) [reply lib/V8/v8-shell.cpp]: We use the steams only in a few places and as you see a function from basics was used...

Comment (ObiWahn) [reply lib/V8/v8-shell.cpp]: Again more macros and legacy c functions are not the solution. Lets wait for c++17 support. Then we should refactor the filesystem related functions with std::filesystem. As I said in the retro: "Can we please distinguish between fixes and beautification. This PR is about testing and fixing the code so that one of the tests works. What is next for this PR? Maybe remove the 4 or 5 `#ifdef`s in v8-utils.cpp? How about doing internal API improvement in a dedicated PR? Must we always mix stuff up?

**Comment by ObiWahn** [comment]
http://jenkins01.arangodb.biz:8080/view/Windows/job/arangodb-ANY-windows-matrix/82/

**Comment by dothebart** [comment]
no relevant errors in the single server tests.
